external-crates/move/crates/move-compiler/tests/sui_mode/linter/random_objects.move

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+module a::test1 {
+    use sui::random::{Random, RandomGenerator};
+
+    entry fun should_work(_r: &Random) {}
+
+    public fun not_allowed1(_x: u64, _r: &Random) {}
+    public fun not_allowed2(_rg: &RandomGenerator, _x: u64) {}
+    public fun not_allowed3(_r: &Random, _rg: &RandomGenerator, _x: u64) {}
```

Comment:
We should add a test for `public(friend)` function, just to document how we expect to handle it